operator/tests/e2e/e2e_test.go

```diff
@@ -927,126 +856,73 @@ func setupWorkloadIdentityForNamespace(namespace, serviceAccEmail, projectID str
 	return nil
 }
 
-func getArtifactRegistryRepositorySample(kccReleaseAssetsDir, uniqueID string) (repoName string, repoYAMLDir string, err error) {
-	repoYAMLDir = path.Join(kccReleaseAssetsDir, "samples", "resources", "artifactregistryrepository")
-	yamlPaths, err := getYAMLFilesInDir(repoYAMLDir)
-	if err != nil {
-		return "", "", fmt.Errorf("error getting paths to YAML files in ArtifactRegistryRepository sample directory '%v': %w", repoYAMLDir, err)
-	}
-	for _, yamlPath := range yamlPaths {
-		b, err := ioutil.ReadFile(yamlPath)
-		if err != nil {
-			return "", "", fmt.Errorf("error reading file '%v': %w", yamlPath, err)
-		}
-		s := string(b)
-		s = strings.ReplaceAll(s, "sample", "sample"+uniqueID)
-		s = strings.ReplaceAll(s, "dep", "dep"+uniqueID)
-
-		// Write back modified YAML to disk
-		if err := writeToFile(s, yamlPath); err != nil {
-			return "", "", fmt.Errorf("error updating file '%v': %w", yamlPath, err)
-		}
-	}
-	repoName, err = getRepoNameFromArtifactRegistryRepositorySampleDir(repoYAMLDir)
-	if err != nil {
-		return "", "", fmt.Errorf("error getting name of ArtifactRegistryRepository for ArtifactRegistryRepository sample directory '%v': %w", repoYAMLDir, err)
-	}
-	return repoName, repoYAMLDir, nil
-}
-
-func getRepoNameFromArtifactRegistryRepositorySampleDir(repoYAMLDir string) (string, error) {
-	unstructs := make([]*unstructured.Unstructured, 0)
-	yamlPaths, err := getYAMLFilesInDir(repoYAMLDir)
-	if err != nil {
-		return "", fmt.Errorf("error getting paths to YAML files in directory '%v': %w", repoYAMLDir, err)
-	}
-	for _, yamlPath := range yamlPaths {
-		u, err := utils.ReadFileToUnstructs(yamlPath)
-		if err != nil {
-			return "", fmt.Errorf("error converting file '%v' to unstructs: %w", yamlPath, err)
-		}
-		unstructs = append(unstructs, u...)
-	}
-	repoNames := make([]string, 0)
-	for _, u := range unstructs {
-		if u.GetKind() == "ArtifactRegistryRepository" {
-			repoNames = append(repoNames, u.GetName())
-		}
-	}
-	switch len(repoNames) {
-	case 0:
-		return "", fmt.Errorf("no ArtifactRegistryRepository found in directory '%v'", repoYAMLDir)
-	case 1:
-		return repoNames[0], nil
-	default:
-		return "", fmt.Errorf("multiple ArtifactRegistryRepositories found in directory '%v'", repoYAMLDir)
+func getArtifactRegistryRepository(namespace string, uniqueID string) (*unstructured.Unstructured, error) {
```

Comment:
Some non-blocking thoughts:

We may want to somehow match this sample with the latest working sample. In the future we may introduce breaking changes in the CRDs (unlikely) or update CRD version (v1beta1 -> v1), and this hard-coded, secret yaml may only be detected "after release" just like this time.